tensorboard/components/tf_data_selector/tf-data-selector.ts

```diff
@@ -0,0 +1,92 @@
+/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the 'License');
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+namespace tf_data_selector {
+
+Polymer({
+  is: 'tf-data-selector',
+  properties: {
+    experiments: Array,
+
+    experimentColors: {
+      type: Array,
+      value: () => [],
+    },
+
+    runs: Array,
```

Comment:
Should tf-data-selector maintain state about individual runs?  I would have thought it would have state about experiments (used to add new tf-data-selector-row elements for experiments) but it seems like the row elements are designed to maintain all their run state internally, which would mean tf-data-selector itself wouldn't manage that as well?

Reply:
It needs to share the selection state (<exp, runs, tags> tuple) with a wrapping plugin later on. I still have not decided on the right medium/API yet. 

This property is mean to communicate to its parent which is not yet implemented.

Reply:
Okay, sure.  If you leave things like TODOs noting that areas are not yet implemented then it's a bit more obvious when reviewing that some parts of the code may not fully make sense yet.